### PR TITLE
Test changes for simulating OOME reported by a user

### PR DESCRIPTION
**Pull request by swatisawant**
## Changes proposed in this pull request

- Modified the schema in the existing putInto long-running test to include 30 columns including most of the string columns
- In this test, 20 columns will be having valid values and 10 columns will be having null values
- Key_columns is specified by a composite key including two string columns
-  Added insert ops in the test with using data from JSON files at regular intervals.
- Added a job to generate the required number of JSON data files containing unique values for key_columns for a user-specified schema
- Framework changes to remove parameters added explicitly for taking heap dump on OOME
 - Reduced default SleepTimeSecsForJobStatus and SleepTimeSecsForMemberStatus to 5 secs from 120 secs and 30 secs respectively in order to minimize the test execution time

## Patch testing

(Fill in the details about how this patch was tested)

## ReleaseNotes.txt changes

(Does this change require an entry in ReleaseNotes.txt? If yes, has it been added to it?)

## Other PRs 

(Does this change require changes in other projects- store, spark, spark-jobserver, aqp? Add the links of PR of the other subprojects that are related to this change)

### CODE REVIEW

File: dtests/src/resources/scripts/putInto/create_and_load_testLTableWith30Columns.sql [hunk 1]
```diff
@@ -0,0 +1,184 @@
+-- DROP TABLE IF ALREADY EXISTS --
+DROP TABLE IF EXISTS testL1;
+
+create table testL1 (id string NOT NULL, data string, data2 decimal, APPLICATION_ID string NOT NULL, ORDERGROUPID string,
+ PAYMENTADDRESS1 string, PAYMENTADDRESS2 string, PAYMENTCOUNTRY string, PAYMENTSTATUS string, PAYMENTRESULT string,
+ PAYMENTZIP string, PAYMENTSETUP string, PROVIDER_RESPONSE_DETAILS string, PAYMENTAMOUNT string, PAYMENTCHANNEL string,
+  PAYMENTCITY string, PAYMENTSTATECODE string,PAYMENTSETDOWN string,  PAYMENTREFNUMBER string, PAYMENTST string,
+ PAYMENTAUTHCODE string,  PAYMENTID string, PAYMENTMERCHID string, PAYMENTHOSTRESPONSECODE string,PAYMENTNAME string,
+ PAYMENTOUTLETID string, PAYMENTTRANSTYPE string,  PAYMENTDATE string, CLIENT_ID string, CUSTOMERID string)
+using column options (key_columns 'id,APPLICATION_ID', BUCKETS '128', REDUNDANCY '1');
```
Comment (suranjan) [inline dtests/src/resources/scripts/putInto/create_and_load_testLTableWith30Columns.sql]: Have we added column delta in table definitions? If not the putInto sizes need to be higher e.g 10m and above

File: dtests/src/resources/scripts/putInto/create_and_load_testLTableWith30Columns.sql [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+DROP TABLE IF EXISTS testL20;
+create table testL20 (id string NOT NULL, data string, data2 decimal, APPLICATION_ID string NOT NULL, ORDERGROUPID string,
+ PAYMENTADDRESS1 string, PAYMENTADDRESS2 string, PAYMENTCOUNTRY string, PAYMENTSTATUS string, PAYMENTRESULT string,
+ PAYMENTZIP string, PAYMENTSETUP string, PROVIDER_RESPONSE_DETAILS string, PAYMENTAMOUNT string, PAYMENTCHANNEL string,
+  PAYMENTCITY string, PAYMENTSTATECODE string,PAYMENTSETDOWN string,  PAYMENTREFNUMBER string, PAYMENTST string,
+ PAYMENTAUTHCODE string,  PAYMENTID string, PAYMENTMERCHID string, PAYMENTHOSTRESPONSECODE string,PAYMENTNAME string,
+ PAYMENTOUTLETID string, PAYMENTTRANSTYPE string,  PAYMENTDATE string, CLIENT_ID string, CUSTOMERID string)
+using column options (key_columns 'id,APPLICATION_ID', BUCKETS '128', REDUNDANCY '1');
```
Comment (suranjan) [inline dtests/src/resources/scripts/putInto/create_and_load_testLTableWith30Columns.sql]: As all tables are same schema wise, we should change the snappy properties. numBuckets, redundancy some eviction property etc.

File: dtests/src/resources/scripts/putInto/create_and_load_testLTableWith30Columns.sql (hunk 1 above)
Comment (swatisawant) [reply dtests/src/resources/scripts/putInto/create_and_load_testLTableWith30Columns.sql]: No, COLUMN_MAX_DELTA_ROWS not added in table definitions.  We tried to create a table with the same properties user specified. But yes, we do have put into sizes higher than 10 million. The range is between 10 to 200 million for putInto. I can change the DDL to include COLUMN_MAX_DELTA_ROWS set to lower value.

File: dtests/src/resources/scripts/putInto/create_and_load_testLTableWith30Columns.sql (hunk 2 above)
Comment (swatisawant) [reply dtests/src/resources/scripts/putInto/create_and_load_testLTableWith30Columns.sql]: OK, I will add the table definitions with the change in these properties.

File: dtests/src/test/java/io/snappydata/hydra/putInto/concPutIntoWithSelectUsingSnappyJobWith30Columns.conf [hunk 3]
```diff
@@ -0,0 +1,93 @@
+hydra.Prms-testRequirement = "Test to verify concurrent putInto functionality along with concurrent select ops to identify the memory leak.";
+hydra.Prms-testDescription = "
+This test starts the snappy cluster.
+Creates column table using sql script.
+Runs the concurrent putInto and select queries using 1 thread for each op in snappy job.
+In this test, 4 snappy jobs are being run concurrently with different ranges for update and inserts.
+blockSize=100000L,stepSize=10000L : job doing 90k updates and 10k inserts
```
Comment (suranjan) [inline dtests/src/test/java/io/snappydata/hydra/putInto/concPutIntoWithSelectUsingSnappyJobWith30Columns.conf]: We should have some larger putIntos.

File: dtests/src/test/scala/io/snappydata/hydra/putInto/ConcPutIntoWith30Columns.scala [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            " 'PAYMENTCHANNEL' ||'id', " +
+            " 'PAYMENTCITY' ||'id', 'PAYMENTSTATECODE' ||'id', 'PAYMENTSETDOWN' ||'id',  " +
+            " 'PAYMENTREFNUMBER' ||'id', 'PAYMENTST' ||'id', " +
+            "  '', '', '', '', '', '', '', '', '', '' " +
+            s" from range(${stepSize}, ${blockSize})")
+        val df = snSession.table(tableName)
+        df.repartition(1).write.json("/export/shared/QA_DATA/jsonFile/" + stepSize)
+      }
+    }
+       val doQuery = () => Future {
```
Comment (suranjan) [inline dtests/src/test/scala/io/snappydata/hydra/putInto/ConcPutIntoWith30Columns.scala]: Does this thread complete before test is stopped?

**Pull request merged**